pisound-btn/pisound-btn.c

```diff
@@ -330,42 +380,48 @@ static int get_action_script_path(enum action_e action, unsigned arg0, unsigned
 		strncpy(tmp, g_config_path, sizeof(tmp)-1);
 		dirname(tmp);
 		strncat(tmp, "/", sizeof(tmp)-1 - strlen(tmp));
-		strncat(tmp, script, sizeof(tmp)-1 - strlen(tmp));
+		strncat(tmp, action_script, sizeof(tmp)-1 - strlen(tmp));
 		tmp[sizeof(tmp)-1] = '\0';
 
-		strncpy(dst, tmp, n-1);
-		dst[n-1] = '\0';
+		strncpy(script, tmp, n-1);
+		script[n-1] = '\0';
 	}
 
 	strncpy(args, arguments, n-1);
 	args[n-1] = '\0';
 
-	return strlen(dst);
+	return 0;
```

Comment:
It should return the length of the string as it did before, as it's used by the calling function, to know where to append the click / hold arguments. Otherwise, it tries to invoke " 1" (just the arguments) for example for single click.